src/frontend/src/flows/manage/deviceSettings.ts

```diff
@@ -178,6 +157,69 @@ const deleteDevice = async ({
   }
 };
 
+/* Reset the device and return to caller */
+const resetPhrase = async ({
+  userNumber,
+  connection,
+  device,
+  reload,
+}: {
+  userNumber: bigint;
+  connection: AuthenticatedConnection;
+  device: DeviceData & RecoveryPhrase;
+  reload: () => void;
+}) => {
+  const confirmed = confirm(
+    "Reset your recovery phrase\n\nWas your recovery phrase compromised? Delete your recovery phrase and generate a new one by confirming."
+  );
+  if (!confirmed) {
+    return;
+  }
+
+  // Create a new recovery phrase
+  const recoveryPhrase = generate().trim();
+  const recoverIdentity = await fromMnemonicWithoutValidation(
+    recoveryPhrase,
+    IC_DERIVATION_PATH
+  );
+
+  // If the phrase is protected, prompt for the phrase
+  const newConnection = isProtected(device)
+    ? await deviceConnection(
+        connection,
+        userNumber,
+        device,
+        "Please input your recovery phrase to reset it."
+      )
+    : connection;
+
+  if (newConnection === null) {
+    // User aborted, just return
+    reload();
+    return;
+  }
+
+  // Save the old pubkey (used as index for replace)
+  const oldKey = device.pubkey;
+  device.pubkey = Array.from(
+    new Uint8Array(recoverIdentity.getPublicKey().toDer())
+  );
+
+  try {
+    await withLoader(() => newConnection.replace(oldKey, device));
```

Comment:
This removes the recovery phrase. Now, if you did recover using this recovery phrase and then reset it, it will cause an error _after_ the reset because the user is no longer authenticated to display their devices. I.e. this:
![Screenshot 2023-03-02 at 13 11 16](https://user-images.githubusercontent.com/94825501/222425573-3a93b926-b2c4-45e0-9e8e-1b28d70bd2f8.png)

It would be nice to have also the "you will be logged out warning" here and then show the home screen instead.

Reply:
Good catch! Since we have the key locally we may simply want to use that as the connection. Will give this a think.

Reply:
Ah, that's a possibility too of course. 👍

Reply:
@frederikrothenberger I've changed it to authenticate the user with the _new_ phrase iff the user was authenticated with the recovery phrase. PTAL!